cl_manycore/libraries/bsg_manycore_eva.cpp

```diff
@@ -354,11 +354,254 @@ int default_eva_to_npa(const hb_mc_config_t *cfg,
 	if(default_eva_is_local(eva))
 		return default_eva_to_npa_local(cfg, origin, src, eva, npa, sz);
 
-	bsg_pr_err("%s: EVA 0x%x did not map to a known region\n", 
+	bsg_pr_err("%s: EVA 0x%x did not map to a known region\n",
 		hb_mc_eva_addr(eva), __func__);
 	return HB_MC_FAIL;
 }
 
+/**
+ * Check if a DRAM EPA is valid.
+ * @param[in] config  An initialized manycore configuration struct
+ * @param[in] npa     An npa to translate
+ * @param[in] tgt     Coordinates of the target tile
+ * @return true if the EPA is valid, false otherwise.
+ */
+static bool default_dram_epa_is_valid(const hb_mc_config_t *cfg,
+                                      hb_mc_epa_t epa,
+                                      const hb_mc_coordinate_t *tgt)
+{
+        return epa < hb_mc_config_get_dram_size(cfg);
+}
+
+
+/**
+ * Check if a local EPA is valid.
+ * @param[in] config  An initialized manycore configuration struct
+ * @param[in] npa     An npa to translate
+ * @param[in] tgt     Coordinates of the target tile
+ * @return true if the EPA is valid, false otherwise.
+ */
+static bool default_local_epa_is_valid(const hb_mc_config_t *config,
+                                       hb_mc_epa_t epa,
+                                       const hb_mc_coordinate_t *tgt)
+{
+        hb_mc_epa_t floor = HB_MC_TILE_EPA_DMEM_BASE;
+        hb_mc_epa_t ceil  = HB_MC_TILE_EPA_DMEM_BASE + hb_mc_config_get_dmem_size(config);
+        return (epa >= floor) && (epa < ceil);
+}
+
+/**
+ * Get the Y-coordinate of DRAM.
+ * @param[in]  config  An initialized manycore configuration struct
+ * @return The Y-coordinate of DRAM.
+ */
+static hb_mc_idx_t default_get_dram_y(const hb_mc_config_t *config)
+{
+        hb_mc_coordinate_t dims = hb_mc_config_get_dimension(config);
+        return hb_mc_coordinate_get_y(dims)+1;
+}
+
+/**
+ * Check if an NPA is a host DRAM.
+ * @param[in] config  An initialized manycore configuration struct
+ * @param[in] npa     An npa to translate
+ * @param[in] tgt     Coordinates of the target tile
+ * @return true if the NPA is DRAM, false otherwise.
+ */
+static bool default_npa_is_dram(const hb_mc_config_t *config,
+                                const hb_mc_npa_t *npa,
+                                const hb_mc_coordinate_t *tgt)
+{
+        return (hb_mc_npa_get_y(npa) == default_get_dram_y(config))
```

Comment:
I don't think the location of dram should be associated with default - I think it should be associated with the configuration